Userland/Services/WebDriver/Session.cpp

```diff
@@ -232,6 +233,187 @@ ErrorOr<void, Variant<HttpError, Error>> Session::delete_window()
+    auto location_strategy = using_;
+
+    // 3. Let selector be equal to value.
+    auto selector = value;
+
+    // 4. Let elements returned be the result of trying to call the relevant element location strategy with arguments start node, and selector.
+    auto location_strategy_handler = s_locator_strategies.first_matching([&](LocatorStrategy const& match) { return match.name == location_strategy; });
+    if (!location_strategy_handler.has_value())
+        return HttpError { 400, "invalid argument", "No valid location strategy" };
+
+    auto elements_or_error = (this->*location_strategy_handler.value().handler)(start_node, selector);
+
+    // 5. If a DOMException, SyntaxError, XPathException, or other error occurs during the execution of the element location strategy, return error invalid selector.
+    if (elements_or_error.is_error())
+        return HttpError { 400, "invalid selector", String::formatted("The location strategy could not finish: {}", elements_or_error.release_error().message) };
+
+    auto elements = elements_or_error.release_value();
+
+    // FIXME: 6. If elements returned is empty and the current time is less than end time return to step 4. Otherwise, continue to the next step.
+    (void)end_time;
+
+    // 7. Let result be an empty JSON List.
+    auto result = JsonArray();
+
+    // 8. For each element in elements returned, append the web element reference object for element, to result.
+    for (auto const& element : elements) {
+        result.append(JsonValue(web_element_reference_object(element)));
+    }
+
+    // 9. Return success with data result.
+    return result;
+}
+
+// https://w3c.github.io/webdriver/#dfn-table-of-location-strategies
+Vector<Session::LocatorStrategy> Session::s_locator_strategies = {
+    { "css selector", &Session::locator_strategy_css_selectors },
+    { "link text", &Session::locator_strategy_link_text },
+    { "partial link text", &Session::locator_strategy_partial_link_text },
+    { "tag name", &Session::locator_strategy_tag_name },
+    { "xpath", &Session::locator_strategy_x_path },
+};
+
+// https://w3c.github.io/webdriver/#css-selectors
+ErrorOr<Vector<Session::LocalElement>, HttpError> Session::locator_strategy_css_selectors(Session::LocalElement const& start_node, StringView const& selector)
+{
+    // 1. Let elements be the result of calling querySelectorAll() with start node as this and selector as the argument.
+    //    If this causes an exception to be thrown, return error with error code invalid selector.
+    auto elements_ids = m_browser_connection->query_selector_all(start_node.id, selector);
+
+    if (!elements_ids.has_value())
+        return HttpError { 400, "invalid selector", "query_selector_all returned failed!" };
+
+    Vector<Session::LocalElement> elements;
+    for (auto id : elements_ids.release_value()) {
+        elements.append({ id });
+    }
+
+    // 2.Return success with data elements.
+    return elements;
```

Comment:
I'm not 100% sure but I think you need to `return move(elements)` since it's effectively calling the `ErrorOr` constructor with `elements` as an argument?